app/data/get-completed-payment-request-by-reference-id.js

```diff
@@ -1,4 +1,4 @@
-const db = require('../data')
+const db = require('.')
```

Comment:
This is quite subjective and depends on the complexity of the app and what those queries are actually doing.  

If the queries are common across multiple business logic modules then makes sense for them to be centralised somewhere, in the data folder or otherwise.

But if they start to become diverse, then building up a collection of specific business logic queries away from the the rest of the actual logic becomes painful.  Essentially this becomes a dumping ground of queries.

As we're using sequelize, which is an abstraction over the connection, I'd say that adds to the argument to keep them with the business logic they're running.

If we weren't using sequelize and were running raw queries, then maybe that's also an argument to centralise a lot of the connection pooling and santization.

I think these queries we have are pretty specific to particular capability so personally I'd have them grouped with that functionality instead.

Also makes it easier if we need to decouple a service if it's doing too much as moving modules as entities to another application is simpler.

Reply:
I agree with those points.

Only slight kink in that is - possibly for some of these queries - what if 2 separate business logic files run the same query? Where does the query code reside then? I _think_ I've seen this in other services and it sometimes confuses or takes me longer to realise, 'oh, that query is in another folder/place as its reused'. Especially if that other business logic (I'm thinking I've poss'ly seen it in processing) is under quite a large subfolder structure. I think this gives a fake psuedo coupling appearance between that sub-folder file and the other file as in its code and test files you're accessing code from it.

There is no solution for this and for 99% of the time the queries are unique for the business logic so happy to revert.